### ISSUE TITLE
Stop using Read the Docs

### ISSUE THREAD

**Issue by Kodiologist** [hylang/hy#2417]
Having ads in the manual is extraordinarily tacky. We should probably just host the web versions of Hy and Hyrule's manuals on Arfer.net, where I also host [the new Hylang.org](http://hylang.org). For simplicity, we can serve only the stable release of the manual. We would then just rebuild it as part of the release process.

**Comment by chiefnoah** [comment]
Why not just use gitlab pages? It's free and supports arbitrary static sites

**Comment by chiefnoah** [comment]
Oh oops, I typoed "GitLab". Meant GitHub Pages. It's an optional feature on each repo

**Comment by Kodiologist** [comment]
I don't think GitHub Pages is a great fit for any static site where the pages are automatically generated, as with Sphinx in the case of our documentation. GitHub only serves what's committed—it can't run a program to generate the pages to serve—so all the generated pages would have to be committed.

**Comment by scauligi** [comment]
So it looks like there are a few different [community plugins](https://github.com/marketplace?type=actions&query=Sphinx) for using GitHub Actions to automatically build Sphinx documentation, for example [this one](https://github.com/marketplace/actions/sphinx-pages) which looks like it should be able to build+deploy docs to GitHub Pages, maybe worth looking into?

**Comment by Kodiologist** [comment]
Hmm, perhaps that sort of use of GitHub Pages is more viable than I thought.

**Comment by jquast** [comment]
You can support rtd by paying $5/mo, and in return you can remove ads from 3 projects

**Comment by atisharma** [comment]
I don't much mind where the docs are hosted, so long as the "bus problem" is planned for.

This happened in Void linux, for example, when xtraem disappeared and nobody had the keys.

**Comment by Kodiologist** [comment]
Yeah, that kind of thing can be painful. For Hy there's some protection in that both Paul and I have most of the credentials.

**Comment by goyalyashpal** [comment]
* current hylang docs:
https://docs.hylang.org/

here's how helix does it:

- site:
https://docs.helix-editor.com/master/print.html
- repo:
https://github.com/helix-editor/helix/tree/master/book/src?search=1
- service used:
_couldn't find on site_

and wezterm:
- site:
https://wezfurlong.org/wezterm/
- repo: 
https://github.com/wez/wezterm/tree/main/docs/index.md
- service used:
https://squidfunk.github.io/mkdocs-material/

**Comment by Kodiologist** [comment]
Read the Docs gave me a lot of grief today while I was releasing Hy 0.27.0. Its implicit default dependencies had some weird interaction with the dependencies we ask for, which led to `pip` spending an hour in fruitless search to satisfy impossible requirements. I kept trying to cancel documentation-building jobs and the cancellation request would be ignored or jobs would spontaneously reactivate. Even if things had worked as-is, Read the Docs is making a breaking change requiring new configuration files this September. I added the configuration file and got things working again, but overall, my appetite for escaping from Read the Docs has certainly increased.

**Comment by thomasmatecki** [comment]
This seems pretty straightforward, see #2485: https://thomasmatecki.github.io/hy/. 
There are some changes to the repository settings that need to happen: https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site

The "book" theme is an experiment. I'll can switch back to the read the docs theme if we want to go ahead this this.

Also there will be links to track down and fix.

**Comment by goyalyashpal** [comment]
> Its implicit default dependencies had some weird interaction with the dependencies we ask for, which led to `pip` spending an hour in fruitless search to satisfy impossible requirements.

do you think [micromamba]'s powerful dependency solver could have helped with that?

[micromamba]: https://mamba.readthedocs.io

**Comment by Kodiologist** [comment]
No idea. I find packaging and dependency-management issues annoying and so I do the minimum.

**Comment by Kodiologist** [comment]
I've picked this up recently. My approach is to integrate the process of building the Hy and Hyrule manuals into [hyhomepage](https://github.com/hylang/hyhomepage) and host them the same way as the rest of hylang.org.
